### ISSUE TITLE
App sync error: `expected a string but was BEGIN_ARRAY`

### ISSUE THREAD

**Issue by GeorgeHahn** [wallabag/wallabag#4411]
### Issue details

Part of the way through initial sync on my device, the sync process halts and an error is shown in the notification drawer. The sync process does not continue and no items are shown.

Error:
```
Error: com.squareup.moshi.JsonDataException: Expected a string but was BEGIN_ARRAY at path $._embedded.items[4].published_by[0]
```

### Environment

Wallabag version: 2.3.8
Installation method: Docker (running latest tag, `f6b2b6b2a1fd`)
Running on a local network server
Database: SQLite

### Steps to reproduce/test case

Sync with android app version 2.3.0 (219), observe failure midway through sync.

**Comment by GeorgeHahn** [comment]
I followed the app sync to find the item causing this failure. Here's the entry where the error occurs (many fields omitted):

```jsonc
{
    "page": 30,
    "limit": 30,
    "_embedded": {
        "items": [
            {
                // this is items[4]:
                "url": "https://www.newyorker.com/books/under-review/the-bullshit-job-boom",
                "published_by": [
                    [
                        "Nathan Heller"
                    ]
                ]
            }
        ]
    }
}
```

**Comment by anarcat** [comment]
i am also getting this error on full sync, which is a big problem because it breaks sync for new devices completely... here is my backtrace:

https://paste.anarc.at/publish/2020-09-09-RbW8ARAMnriRw4H0UC6lLhQIigsd7HDkT2OpaxUDSf0/stdin.txt

it would be nice if wallabag at least handled that exception correctly...

**Comment by di72nn** [comment]
[This comment](https://github.com/wallabag/android-app/issues/887#issuecomment-570032818) explains how to fix this.

**Comment by anarcat** [comment]
yeah, but that requires server access, which people might not necessarily have

**Comment by di72nn** [comment]
Self-hosting users should be able to deal with it themselves. Users of hosted solutions (wallabag.it, etc.) should ask server administrator to deal with it. Am I missing some other category?

**Comment by anarcat** [comment]
i guess my argument is that even server admins might not be aware of the problem or might not clearly see how to fix it, because it's not part of release notes or documented other than in issues such as this one.

i'll also note that this issue is filed against the server side, and still open. I would argue it should be fixed by shipping a database migration that corrects this problem and https://github.com/wallabag/android-app/issues/890

**Comment by j0k3r** [comment]
Both bug should be fixed with 2.4.0 I guess. Can someone confirm?

**Comment by solkaz** [comment]
@j0k3r @di72nn I still am encountering this issue on 2.4.2 (Pixel 3a, Android 11, using app.wallabag.it)

**Comment by di72nn** [comment]
> I still am encountering this issue

It should be fine if you refetch the article.

I believe fetching is fixed, but the old data may still be broken.

**Comment by nicosomb** [comment]
@GeorgeHahn @di72nn hello. Could you try with latest wallabag version and tell us if this bug still occurs please?

**Comment by GeorgeHahn** [comment]
Sorry, I don't use Wallabag anymore and will be unable to test this.

**Comment by virtadpt** [comment]
Still broken for me.
